exchanges/okx/okx.go

```diff
@@ -4056,152 +5240,131 @@ func (ok *Okx) GetOpenInterestAndVolumeExpiry(ctx context.Context, currency stri
-	}
-	putblockvol, err := strconv.ParseFloat(resp[6], 64)
+	var resp [7]types.Number
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getTakerFlowEPL, http.MethodGet, common.EncodeURLValues("rubik/stat/option/taker-block-volume", params), nil, &resp, false, true)
 	if err != nil {
 		return nil, err
 	}
 	return &CurrencyTakerFlow{
-		Timestamp:       time.UnixMilli(timestamp),
-		CallBuyVolume:   callbuyvol,
-		CallSellVolume:  callselvol,
-		PutBuyVolume:    putbutvol,
-		PutSellVolume:   putsellvol,
-		CallBlockVolume: callblockvol,
-		PutBlockVolume:  putblockvol,
+		Timestamp:       time.UnixMilli(resp[0].Int64()),
+		CallBuyVolume:   resp[1].Float64(),
+		CallSellVolume:  resp[2].Float64(),
+		PutBuyVolume:    resp[3].Float64(),
+		PutSellVolume:   resp[4].Float64(),
+		CallBlockVolume: resp[5].Float64(),
+		PutBlockVolume:  resp[6].Float64(),
 	}, nil
 }
 
+// ********************************************************** Affiliate **********************************************************************
+
+// The Affiliate API offers affiliate users a flexible function to query the invitee information.
+// Simply enter the UID of your direct invitee to access their relevant information, empowering your affiliate business growth and day-to-day business operation.
+// If you have additional data requirements regarding the Affiliate API, please don't hesitate to contact your BD.
+// We will reach out to you through your BD to provide more comprehensive API support.
+
+// GetInviteesDetail retrieves affiliate invitees details.
+func (ok *Okx) GetInviteesDetail(ctx context.Context, uid string) (*AffilateInviteesDetail, error) {
+	if uid == "" {
+		return nil, errUserIDRequired
+	}
+	var resp *AffilateInviteesDetail
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getAffilateInviteesDetailEPL, http.MethodGet, "affiliate/invitee/detail?uid="+uid, nil, &resp, true)
+}
+
+// GetUserAffilateRebateInformation this endpoint is used to get the user's affiliate rebate information for affiliate.
+func (ok *Okx) GetUserAffilateRebateInformation(ctx context.Context, apiKey string) (*AffilateRebateInfo, error) {
+	if apiKey == "" {
+		return nil, errInvalidAPIKey
+	}
+	var resp *AffilateRebateInfo
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getUserAffilateRebateInformationEPL, http.MethodGet, "users/partner/if-rebate?apiKey="+apiKey, nil, &resp, true)
+}
+
 // SendHTTPRequest sends an authenticated http request to a desired
 // path with a JSON payload (of present)
 // URL arguments must be in the request path and not as url.URL values
-func (ok *Okx) SendHTTPRequest(ctx context.Context, ep exchange.URL, f request.EndpointLimit, httpMethod, requestPath string, data, result interface{}, authenticated bool) (err error) {
+func (ok *Okx) SendHTTPRequest(ctx context.Context, ep exchange.URL, f request.EndpointLimit, httpMethod, requestPath string, data, result interface{}, authenticated bool, useAsItIs ...bool) (err error) {
```

Comment:
Suggestion: Make a specific type or variable for the `authenticated bool` just so we can have `Authenticated` or `Public` instead of true or false. But up to you if you want to do that.

Reply:
This way it is simple ... but your approach is more **meaningful**, makes the call **less prone to mistake**, can help include different authentication mechanisms(like starkEx), and could be used by other exchanges too. So, I will create these options in the **_order_** folder and make it open to use.

Reply:
But how about the **_AuthType_** type?
It is already implemented and being used by so many exchanges

Reply:
Anyways, I am using the UnauthenticatedRequest and AuthenticatedRequest values of type AuthType in the request package in this implementation as changing these could affect other exchanges.

Reply:
The changes you are speaking of may be impacting `exchange_wrapper_standards`. You need to classify auth requests as auth or it will error